### DISCUSSION TITLE
critic.py - What is the critic markup putting back into the document in the Postprocessor?

### DISCUSSION THREAD

**Discussion by bakoontz** [facelessuser/pymdown-extensions#2538] [Q&A] [answered]
Trying to learn how markdown extensions work, and I'm using critic.py as my example.  I understand most of all what is going on, but what is the Postprocessor doing?  It's not clear to me what, exactly, is being injected back into the HTML stream...

**Comment by facelessuser** [comment, marked as answer]
The postprocessor is used to inject a visual representation of the critic markups. This is a cumbersome process, to pull off well with the Python Markdown, but it works to some degree.  The documentation shows examples. The critic extension is more useful for just accepting or rejecting the changes.

**Comment by bakoontz** [comment]
Thanks for the explanation, makes more sense to me now.On Nov 29, 2024, at 5:09 PM, Isaac Muse ***@***.***> wrote:﻿
The postprocessor is used to inject a visual representation of the critic markups. This is a cumbersome process, to pull off well with the Python Markdown, but it works to some degree.  The documentation shows examples. The critic extension is more useful for just accepting or rejecting the changes.
